tardis/visualization/tools/convergence_plot.py

```diff
@@ -0,0 +1,388 @@
+                title=r"$\mbox{Luminosity}~(erg~s^{-1})$",
+                title_font_size=13,
+                automargin=True,
+                tickmode="auto",
+                nticks=7,
+            ),
+            yaxis3=dict(
+                exponentformat="e",
+                title=r"$~~\mbox{Residual}\\\mbox{Luminosity(%)}$",
+                title_font_size=12,
+                automargin=True,
+                tickformat="%",
+                tickmode="auto",
+                nticks=4,
+            ),
+            legend_tracegroupgap=0,
+            height=630,
+            hoverlabel_align="right",
+            legend_title_text="Luminosity",
+            hoverlabel_font_color="white",
+            margin=dict(b=25, t=25, pad=0),
+        )
+
+        # allows overriding default layout
+        if hasattr(self, "luminosity_plot_config"):
+            self.override_plot_parameters(fig, self.luminosity_plot_config)
+
+        self.luminosity_plot = fig
+
+    def override_plot_parameters(self, fig, parameters):
+        """
+        Overrides default plot properties.
+
+        Parameters
+        ----------
+        fig : go.FigureWidget
+            FigureWidget object to be updated
+        parameters : dict
+            Dictionary used to update the default plot style. The dictionary should have a structure
+            like that of go.FigureWidget.to_dict(), for more information please see https://plotly.com/python/figure-structure/
+        """
+        for key, value in parameters.items():
+            if key == "data":
+                for trace in list(fig.data):
+                    self.override_plot_parameters(trace, value)
+            else:
+                if type(value) == dict:
+                    self.override_plot_parameters(fig[key], value)
+                else:
+                    fig[key] = value
+
+    def build(self, display_plot=True):
+        """
+        Creates empty plasma and luminosity convergence plots and displays them together.
+
+        Parameters
+        ----------
+        display_plot : bool
+            Displays empty plots. Defaults to True.
+        """
+        self.create_plasma_plot()
+        self.create_luminosity_plot()
+        if display_plot:
+            display(
+                widgets.VBox(
+                    [self.plasma_plot, self.luminosity_plot],
+                )
+            )
+
+    def update_plasma_plots(self):
+        """
+        Updates plasma convergence plots every iteration.
+        """
+        x = self.iterable_data["velocity"].to(u.km / u.s).value.tolist()
+        customdata = len(x) * [
+            "<br>"
+            + "Emitted Luminosity: "
+            + f'{self.value_data["Absorbed"][-1]:.2g}'
+            + "<br>"
+            + "Requested Luminosity: "
+            + f'{self.value_data["Requested"][-1]:.2g}'
+            + "<br>"
+            + "Absorbed Luminosity: "
+            + f'{self.value_data["Requested"][-1]:.2g}'
+        ]
+
```

Comment:
Not too sure where the legend stuff is defined, but why is there a Luminosity header for all the lines? This doesn't work if the first line in there is for temperature! Either find a way to separate the temp. line out and then have the Luminosity header for the rest of the lines (and if you do this, remove the luminosity second words from all those labels - e.g. 'emitted luminosity' would become 'emitted') or just remove 
the header. I recall seeing you were finding it difficult to produce 2 separate legends, so perhaps just remove the header.

![Screenshot 2021-06-29 at 15 40 58](https://user-images.githubusercontent.com/47385282/123817859-6a2cbe80-d8f0-11eb-8f2e-64e01e7ac71f.png)

Reply:
done in https://github.com/tardis-sn/tardis/commit/b652101a3ddb66107fe4b73019f63f7834a2a4d7